### PR TITLE
Expose tls.ConnectionState for TLS connections

### PR DESCRIPTION
**Pull request by cbandy**
I wanted to check the TLS version and cipher suite negotiated by Go and RabbitMQ.

### CODE REVIEW

File: tls_test.go [hunk 1]
```diff
@@ -93,6 +93,11 @@ func TestTLSHandshake(t *testing.T) {
 			t.Fatalf("expected to handshake a TLS connection, got err: %v", err)
 		}
 	}
+
+	st := c.ConnectionState()
+	if !st.HandshakeComplete {
+		t.Errorf("expected ConnectionState, got: %+v", st)
```
Comment (michaelklishin) [inline tls_test.go]: This error message doesn't actually warn the user that there was a TLS handshake failure. Please change this to "TLS handshake failed, TLS connection state: …".

File: connection.go [hunk 2]
```diff
@@ -232,6 +236,14 @@ func (me *Connection) LocalAddr() net.Addr {
 	return &net.TCPAddr{}
 }
 
+// ConnectionState returns basic TLS details of the underlying transport.
+func (me *Connection) ConnectionState() tls.ConnectionState {
```
Comment (michaelklishin) [inline connection.go]: This function can confuse the reader slightly: it returns a `tls.ConnectionState` but AMQP 0-9-1 connections also have state and so `*Connection.ConnectionState` doesn't really indicate that what's returned in a TLS upgrade state.

Please rename it to `TLSConnectionState`.

**Comment by michaelklishin** [comment]
Thank you! We'd be happy to include this feature but this specific PR needs a bit of work.

File: tls_test.go (hunk 1 above)
Comment (cbandy) [inline tls_test.go]: Fixed in amended commit.

File: connection.go (hunk 2 above)
Comment (cbandy) [inline connection.go]: Done.

**Pull request merged**